### PR TITLE
[MIRROR] tgui Preferences Menu + total rewrite of the preferences backend

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/61313
--------------------

## About The Pull Request

Rewrites the entire preferences menu in tgui. Rewrites the entire backend to be built upon datumized preferences, rather than constant additions to the preferences base datum.

https://user-images.githubusercontent.com/35135081/132632991-fb7d8e73-6b3a-4a7a-9795-b562d665a606.mp4

https://user-images.githubusercontent.com/35135081/132633011-e7d60352-520e-4ec0-b038-6379928137b3.mp4

Splits game preferences into its own window.

Antagonists are now split into their individual rulesets. You can now be a roundstart heretic without signing up for latejoin heretic, as an example.

This iteration matches parity, and provides very little new functionality, but adding anything new will be much easier.

Fixes #60823
Fixes #28907
Fixes #44887
Fixes #59912
Fixes #58458
Fixes #59181

### Major TODOs
- [x] Quirk icons, from @ Fikou (with some slight adjustments from me)
- [x] Lore text, from @ EOBGames (4/6, need moths and then ethereal lore from @ AMonkeyThatCodes)
- [x] Heavy documentation on how one would add new preferences, species, jobs, etc
- [x] A lot of specialized testing so that people's real data don't get corrupted

## Changelog

:cl: Mothblocks, Floyd on lots of the design
refactor: The preferences menu has been completely rewritten in tgui.
refactor: The "Stop Sounds" verb has been moved to OOC.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
pedro

**Comment by Mothblocks** [comment]
wubbli

**Comment by thestubborn** [comment]
image

**Comment by ghost** [comment]
https://user-images.githubusercontent.com/20987591/133340925-45f9fa9a-f03e-4e12-a271-fc02dde5bb35.mp4

**Comment by KathyRyals** [comment]
Godspeed to the Skyrat coders. May your suffering be quick.

**Comment by lyricalpaws** [comment]
> 
> 
> Godspeed to the Skyrat coders. May your suffering be quick.

we all know it's gonna be slow and painful

**Comment by maxymax13** [comment]
:trollface:

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -309,7 +309,20 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	READ_FILE(S["randomise"],  randomise)
 	READ_FILE(S["persistent_scars"] , persistent_scars)
 
-	READ_FILE(S["loadout_list"] , loadout_list)//SKYRAT EDIT ADDITION
+	//SKYRAT EDIT ADDITION
+	READ_FILE(S["loadout_list"] , loadout_list)
```
Comment (Mothblocks) [inline code/modules/client/preferences_savefile.dm]: you know you can just `/datum/preferences/load_character` and `..()` right? you can make this modular

Comment (Gandalf2k15) [reply code/modules/client/preferences_savefile.dm]: Oh. True. I'll do that, thanks

Comment (Mothblocks) [reply code/modules/client/preferences_savefile.dm]: you guys do this a LOOOOT ive noticed this is just the first time ive felt the need to point it out

**Comment by axietheaxolotl** [comment]
https://user-images.githubusercontent.com/86872535/133381394-9eb2113e-8a27-48cd-b2fc-899c45ab8f37.mp4

File: code/modules/client/preferences_savefile.dm (hunk 1 above)
Comment (Gandalf2k15) [reply code/modules/client/preferences_savefile.dm]: Sometimes we want things to conflict to make sure the thing still works

**Comment by ChungusGamer666** [comment]
Suffer well

https://user-images.githubusercontent.com/82850673/133722383-dbf8a748-811a-40ff-aeba-c91e0efa1375.mp4

File: modular_skyrat/modules/loadouts/readme.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Title: MrMelbert's loadout system!
+
+MODULE ID: LOADOUT
+
+### Description:
+
+A loadoud system that uses TGUI to equip players post roundstart.
```
Comment (ORCACommander) [inline modular_skyrat/modules/loadouts/readme.md]: Loadout

**Comment by IgiariValkyr** [comment]
Looks like no one looked and saw the "Byond Premium for more character slots" code.

**Comment by Gandalf2k15** [comment]
Full merging, most bugs are fixed, the server will be placed in limbo on the old code until this is sorted fully.
war-games-nuclear-launch

**Pull request merged**